tools/autograd/templates/Functions.cpp

```diff
@@ -425,6 +425,16 @@ Tensor cumsum_backward(const Tensor & x, int64_t dim) {
   return ret;
 }
 
+Tensor cummax_backward(const Tensor &grad, const Tensor &input, int64_t dim) {
+  if (input.dim() == 0) {
+    return input;
+  }
+  auto result = at::native::cummax(input, dim);
+  auto ret = std::get<0>(result);
+  auto indices = std::get<1>(result);
+  return grad * at::native::scatter(ret, dim, indices, at::ones(ret.sizes()));
```

Comment:
The `scatter` here is out of place right?

Reply:
yup